underfs/oss/src/main/java/alluxio/underfs/oss/OSSActiveSyncProvider.java

```diff
@@ -0,0 +1,356 @@
+      newFileName = newFileName.substring(1);
+    }
+    if (newFileName.endsWith("/")) {
+      newFileName = newFileName.substring(0, newFileName.length() - 1);
+    }
+    return newFileName;
+  }
+
+  private boolean isChanged(AlluxioURI ufsUri, long ufsLastModifiedTimestamp) {
+    Long lastModified = mUfsUriLastModified.get(ufsUri);
+    if (null == lastModified || lastModified != ufsLastModifiedTimestamp) {
+      return true;
+    }
+    return false;
+  }
+
+  private void updateUriLastModified(AlluxioURI ufsUri, long ufsLastModifiedTimestamp) {
+    mUfsUriLastModified.put(ufsUri, ufsLastModifiedTimestamp);
+  }
+
+  /**
+   * start polling thread.
+   *
+   * @param txId transaction id to start monitoring
+   * @return true if polling thread started successfully
+   */
+  public boolean startPolling(long txId) throws IOException {
+    if (mPollingThread == null) {
+      LOG.info("Start polling from event txID {}", txId);
+      mPollingThread = mExecutorService.scheduleAtFixedRate(() -> updateChangedFiles(), 0,
+          mActiveUfsSyncInterval, TimeUnit.MILLISECONDS);
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * stop polling thread.
+   *
+   * @return true if polling thread stopped successfully
+   */
+  public boolean stopPolling() {
+    if (mPollingThread != null) {
+      mPollingThread.cancel(true);
+      mPollingThread = null;
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * startSync on a ufs uri.
+   *
+   * @param ufsUri the ufs uri to monitor for sync
+   */
+  public void startSync(AlluxioURI ufsUri) {
+    LOG.debug("Add {} as a sync point", ufsUri);
+    mUfsUriList.add(ufsUri);
+  }
+
+  /**
+   * stop sync on a ufs uri.
+   *
+   * @param ufsUri the ufs uri to stop monitoring for sync
+   */
+  public void stopSync(AlluxioURI ufsUri) {
+    LOG.debug("attempt to remove {} from sync point list", ufsUri);
+    mUfsUriList.remove(ufsUri);
+  }
+
+  /**
+   * get the last transaction id.
+   *
+   * @return the last transaction id
+   */
+  private long getLastTxId() {
+    if (mTxIdMap.isEmpty()) {
+      return mCurrentTxId;
+    } else {
+      return Collections.min(mTxIdMap.values());
+    }
+  }
+
+  /**
+   * Get Active SyncInfo from the sync provider.
+   *
+   * @return SyncInfo a syncInfo containing information about what to sync and how to sync
+   */
+  public SyncInfo getActivitySyncInfo() {
+    // The overview of this method is
+    // 1. setup a source of event
+    // 2. Filter based on the paths associated with this mountId
+    // 3. Build History for each of the syncPoint
+    // 4. If heuristics function returns sync, then we sync the syncPoint
+
+    if (mPollingThread == null) {
+      return SyncInfo.emptyInfo();
+    }
+
+    Map<AlluxioURI, Set<AlluxioURI>> syncPointFiles = new HashMap<>();
+    long txId = 0;
+    try (LockResource r = new LockResource(mWriteLock)) {
+      initNextWindow();
+      if (mEventMissed) {
```

Comment:
I see that `mEventMissed = false` in the default constructor method, when will it be true?

Reply:
Thanks. There is no need to use this member "mEventMissed". I am going to delete this member.